packages/tools/src/tools/segmentation/ThresholdPreviewTool.ts

```diff
@@ -0,0 +1,235 @@
+
+    const { volumeId } = representationData[type] as LabelmapSegmentationData;
+
+    const segmentation = cache.getVolume(volumeId);
+
+    const segmentIndex =
+      segmentIndexController.getActiveSegmentIndex(segmentationId);
+
+    const operationData = {
+      points: [],
+      imageVolume,
+      volume: segmentation,
+      segmentationId,
+      segmentIndex,
+      segmentsLocked: [],
+      strategySpecificConfiguration:
+        this.configuration.strategySpecificConfiguration,
+    };
+
+    this.applyActiveStrategy(null, operationData);
+
+    function _getSegmentationRepresentation(segmentationId, toolGroupId) {
+      const segmentationRepresentations =
+        segmentationState.getSegmentationRepresentations(toolGroupId);
+
+      if (segmentationRepresentations.length === 0) {
+        return;
+      }
+
+      // Todo: this finds the first segmentation representation that matches the segmentationId
+      // If there are two labelmap representations from the same segmentation, this will not work
+      const representation = segmentationRepresentations.find(
+        (representation) => representation.segmentationId === segmentationId
+      );
+
+      return representation;
+    }
+
+    const segmentationRepresentation = _getSegmentationRepresentation(
+      segmentationId,
+      this.toolGroupId
+    );
+
+    const animationLength = 750;
+
+    // Animation for the segmentIndex
+
+    const { fillAlpha } = this.getConfiguration();
+
+    let count = 0;
+    const intervalTime = 16;
+    const numberOfFrames = Math.ceil(animationLength / intervalTime);
+
+    clearInterval(this.highlightIntervalId);
+
+    this.highlightIntervalId = setInterval(() => {
+      const x = (count * intervalTime) / animationLength;
+      const easeOutFillAlpha = easeInOutBell(x, fillAlpha);
+
+      if (easeOutFillAlpha < 0) {
+        count = 0;
+      } else {
+        count++;
+      }
+      segmentationConfig.setSegmentSpecificConfig(
+        this.toolGroupId,
+        segmentationRepresentation.segmentationRepresentationUID,
+        {
+          [segmentIndex]: {
+            LABELMAP: {
+              fillAlpha: easeOutFillAlpha,
+            },
+          },
+        }
+      );
+
+      if (count === numberOfFrames) {
+        segmentationConfig.setSegmentSpecificConfig(
+          this.toolGroupId,
+          segmentationRepresentation.segmentationRepresentationUID,
+          {}
+        );
+      }
+    }, intervalTime);
+  }
+}
+
+export function easeInOutBell(x: number, baseline: number): number {
+  const alpha = 1 - baseline;
+
+  // prettier-ignore
+  if (x < 1 / 4) {
+    return  4 * Math.pow(2 * x, 3) * alpha + baseline;
+  } else if (x < 1 / 2) {
+    return (1 - Math.pow(-4 * x + 2, 3) / 2) * alpha + baseline;
+  } else if (x < 3 / 4) {
+    return (1 - Math.pow(4 * x - 2, 3) / 2) * alpha + baseline;
+  } else {
+    return (- 4 * Math.pow(2 * x - 2, 3)) * alpha + baseline;
+  }
+}
```

Comment:
this is useful, put it outside in utils

Reply:
Moved to the utilities inside animationHelper